backend/copr_backend/storage.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for various data storages, e.g. results directory on backend, Pulp, etc.
+"""
+
+import os
+from copr_backend.helpers import call_copr_repo
+from copr_backend.helpers import run_cmd
+
+
+def storage_for_task(task, opts, log):
+    """
+    Return an appropriate storage object for a given task (build or action)
+    """
+    if task.get("pulp"):
+        return PulpStorage(opts, log)
+    return BackendStorage(opts, log)
+
+
+class Storage:
+    """
+    Storage agnostic, high-level interface for storing and acessing our data
+    """
+
+    def __init__(self, opts, log):
+        self.opts = opts
+        self.log = log
+
+    def createrepo(self, ownername, coprdir, chroot, appstream, devel):
```

Comment:
We can keep this abstractionfor now, but I fear it is not going to work with PULP where we "add RPM to repo", which "generates" version, and there's nothing like a "request for createrepo".

This is one step more on top of the original trivial issue, and I'm not against it.

Reply:
> This is one step more on top of the original trivial issue, and I'm not against it.

I probably misunderstood the scope of #3112. I read the "PULP: create project" as "when a Copr is created, generate a repository in Pulp for it". 

> We can keep this abstractionfor now, but I fear it is not going to work with PULP where we "add RPM to repo", which "generates" version, and there's nothing like a "request for createrepo".

All I know about Pulp is what we have in our research documents, so I wasn't sure if we can map 1:1 all of our current storage operations to what needs to be done in Pulp. This `createrepo` works as expected when creating an empty project, so I thought it is a good enough start, that will be split into multiple methods, and renamed as needed as new Pulp features are being added.

But maybe I can do one more step now and add RPM packages after builds, so the needed abstractions are clearer?